components/net/indexeddb/engines/sqlite.rs

```diff
@@ -0,0 +1,513 @@
+                            },
+                            Err(e) => {
+                                let _ = sender.send(err!(e));
+                            },
+                        }
+                    },
+                    AsyncOperation::ReadWrite(AsyncReadWriteOperation::Clear(sender)) => {
+                        let result = object_data_model::Entity::delete_many()
+                            .filter(object_data_model::Column::ObjectStoreId.eq(object_store.id))
+                            .exec(&conn)
+                            .await;
+                        let _ = match result {
+                            Ok(_) => sender.send(Ok(())),
+                            Err(e) => sender.send(err!(e)),
+                        };
+                    },
+                    AsyncOperation::ReadOnly(AsyncReadOnlyOperation::GetKey {
+                        sender,
+                        key_range,
+                    }) => {
+                        let result = object_data_model::Entity::find()
+                            .filter(object_data_model::Column::ObjectStoreId.eq(object_store.id))
+                            .filter(range_to_query(key_range))
+                            .one(&conn)
+                            .await;
+
+                        match result {
+                            Ok(result) => {
+                                let _ =
+                                    sender
+                                        .send(Ok(result
+                                            .map(|blob| serialize::decode(&blob.key).unwrap())));
+                            },
+                            Err(e) => {
+                                let _ = sender.send(err!(e));
+                            },
+                        }
+                    },
+                }
+            }
+        });
+        rx
+    }
+
+    // TODO: we should be able to error out here, maybe change the trait definition?
+    fn has_key_generator(&self, store_name: SanitizedName) -> bool {
+        HANDLE.block_on(async {
+            if let Some(model) = object_store_model::Entity::find()
+                .filter(object_store_model::Column::Name.eq(store_name.to_string()))
+                .one(&self.connection)
+                .await
+                .unwrap()
+            {
+                model.auto_increment
+            } else {
+                false
+            }
+        })
+    }
+
+    fn key_path(&self, store_name: SanitizedName) -> Option<KeyPath> {
+        HANDLE.block_on(async {
+            if let Some(model) = object_store_model::Entity::find()
+                .filter(object_store_model::Column::Name.eq(store_name.to_string()))
+                .one(&self.connection)
+                .await
+                .unwrap()
+            {
+                model
+                    .key_path
+                    .map(|key_path| bincode::deserialize(&key_path).unwrap())
```

Comment:
Should we use the serialize module instead?

Reply:
See above